### PR TITLE
Add table display parameters

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**

### CODE REVIEW

File: api/plot/utils.py [hunk 1]
```diff
@@ -213,6 +215,8 @@ def table(
         self,
         dataArray: OptionalLists,
         cellWidth: int,
+        display_style: Union[TableDisplayType, None] = None,
```
Comment (PeterC-DLS) [inline api/plot/utils.py:218]: Switch to using `|` instead of `Union` as we're using Python `3.10`

File: client/component/TableDisplay.tsx [hunk 2]
```diff
@@ -5,7 +5,25 @@ import React from 'react';
 
 
 class TableDisplay extends React.Component<TableDisplayProps> {
-  fmt = new Intl.NumberFormat('en', {notation: 'scientific', maximumSignificantDigits: 6});
+
+  displayStyle = (
+    this.props.displayParams != undefined && this.props.displayParams.displayType != undefined?
```
Comment (PeterC-DLS) [inline client/component/TableDisplay.tsx:10]: Add space before `?`

File: api/plot/utils.py (hunk 1 above)
Comment (VictoriaBeilsten-Edmands) [reply api/plot/utils.py]: Or better to have `Optional[TableDisplayType] = None` here?

Comment (PeterC-DLS) [reply api/plot/utils.py]: I think `TableDisplayType | None` is preferred as you don't need the additional import

File: client/component/TableDisplay.tsx [hunk 3]
```diff
@@ -5,7 +5,25 @@ import React from 'react';
 
 
 class TableDisplay extends React.Component<TableDisplayProps> {
-  fmt = new Intl.NumberFormat('en', {notation: 'scientific', maximumSignificantDigits: 6});
+
+  displayStyle = (
+    this.props.displayParams != undefined && this.props.displayParams.displayType != undefined ?
```
Comment (PeterC-DLS) [inline client/component/TableDisplay.tsx:10]: More correct to use `!==`

File: api/plot/custom_types.py [hunk 4]
```diff
@@ -13,6 +13,20 @@ class StatusType(str, Enum):
...
+    """Class for table display type."""
+
+    standard = "standard"
+    scientific = "scientific"
+
+
+class TableDisplayParams(BaseModel):
+    """Class for representing table display type and number of digits."""
+
+    displayType: Optional[TableDisplayType]
```
Comment (PeterC-DLS) [inline api/plot/custom_types.py:26]: Use `|` annotation

File: api/plot/custom_types.py [hunk 5]
```diff
@@ -13,6 +13,20 @@ class StatusType(str, Enum):
...
+    numberDigits: Optional[int]
```
Comment (PeterC-DLS) [inline api/plot/custom_types.py:27]: Add custom validation to ensure this value is good

File: api/plot/custom_types.py [hunk 6]
```diff
@@ -13,6 +13,20 @@ class StatusType(str, Enum):
     busy = "busy"
 
 
+class TableDisplayType(str, Enum):
+    """Class for table display type."""
+
+    standard = "standard"
```
Comment (PeterC-DLS) [inline api/plot/custom_types.py:19]: Add doc to explain what these are

**Comment by PeterC-DLS** [comment]
Needs rebasing to resolve conflicts

File: client/component/main.d.ts [hunk 7]
```diff
@@ -2,6 +2,10 @@ type ScaleType =  "linear" | "log" | "symlog" | "sqrt" | "gamma";
 
 type StatusType =  "ready" | "busy";
 
+type TableDisplayType =  "scientific" | "standard";
+
+type MsgType = "clear_data" | "new_image_data" | "new_line_data" | "new_multiline_data" | "status";
```
Comment (PeterC-DLS) [inline client/component/main.d.ts:7]: Don't need this any longer

File: client/component/main.d.ts [hunk 8]
```diff
@@ -98,6 +108,15 @@ interface HeatmapPlotProps extends ImagePlotProps {
   heatmapScale: ScaleType;
 }
 
+<<<<<<< HEAD
```
Comment (PeterC-DLS) [inline client/component/main.d.ts:111]: Need to edit this further

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**